evalml/pipelines/components/estimators/regressors/baseline_regressor.py

```diff
@@ -38,9 +42,10 @@ def __init__(self, strategy="mean", random_state=0, **kwargs):
     def fit(self, X, y=None):
         if y is None:
             raise ValueError("Cannot fit Baseline regressor if y is None")
-
-        if not isinstance(y, pd.Series):
-            y = pd.Series(y)
+        X = _convert_to_woodwork_structure(X)
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
+        y = _convert_to_woodwork_structure(y)
+        y = _convert_woodwork_types_wrapper(y.to_series())
```

Comment:
Arguable: is it necessary to do this conversion for `X` if all we need is `X.shape`? I did here for consistency but...

Reply:
Ah got it. Yep, may as well use the same code for now for consistency.